### DISCUSSION TITLE
Sensorless homing Integration?

### DISCUSSION THREAD

**Discussion by natetheskate** [jschuh/klipper-macros#165] [Q&A] [answered]
Hi, I've been using your macros a while and they are awesome! Thank you!

I recently converted one of my Voron 2.4's over to full canbus with sensorless homing. When setting up sensorless homing it was required to create a homing.cfg file to modify homing which I believe interferes/overlaps with your macros. Any advise on what to do/modify to make them play well/work together?

I did attempt to modify the lines below referencing "G28 X/Y" to "G28.6245197 X/ G28.6245197 Y" which enables me to print fine however when the print head parks (after a print for example) in the back left corner this creates another problem where the X/Y are at max position and therefore have no room to move if/when a new G28 is triggered. Could I just take out the "G1 Y-10 F1200" and G1 X-10 F1200" ? Thanks for any help!

This is what is listed on vorondesign:

[gcode_macro _HOME_X]
gcode:
    # Always use consistent run_current on A/B steppers during sensorless homing
    {% set RUN_CURRENT_X = printer.configfile.settings['tmc2209 stepper_x'].run_current|float %}
    {% set RUN_CURRENT_Y = printer.configfile.settings['tmc2209 stepper_y'].run_current|float %}
    {% set HOME_CURRENT = 0.7 %}
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={HOME_CURRENT}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={HOME_CURRENT}

    # Home
    G28 X
    # Move away
    G91
    G1 X-10 F1200
    
    # Wait just a second… (give StallGuard registers time to clear)
    G4 P1000
    # Set current during print
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={RUN_CURRENT_X}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={RUN_CURRENT_Y}

[gcode_macro _HOME_Y]
gcode:
    # Set current for sensorless homing
    {% set RUN_CURRENT_X = printer.configfile.settings['tmc2209 stepper_x'].run_current|float %}
    {% set RUN_CURRENT_Y = printer.configfile.settings['tmc2209 stepper_y'].run_current|float %}
    {% set HOME_CURRENT = 0.7 %}
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={HOME_CURRENT}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={HOME_CURRENT}

    # Home
    G28 Y
    # Move away
    G91
    G1 Y-10 F1200

    # Wait just a second… (give StallGuard registers time to clear)
    G4 P1000
    # Set current during print
    SET_TMC_CURRENT STEPPER=stepper_x CURRENT={RUN_CURRENT_X}
    SET_TMC_CURRENT STEPPER=stepper_y CURRENT={RUN_CURRENT_Y}

**Comment by jschuh** [comment, marked as answer]
You'll want to set the following variables in your `[gcode_macro _km_options]` section to leave some distance from the end of the axis for parking positions:

- `variable_park_x` - Defaults to zero; `10` should work
- `variable_park_y` - Defaults to zero; `10` should work
- `variable_start_end_park_y` - Defaults to print_max Y; set it to 10 less than the print max

There are comments explaining each of these variables in [globals.cfg](https://github.com/jschuh/klipper-macros/blob/main/globals.cfg).

**Reply by natetheskate** [reply]
Thank you for the quick response. My apologies for missing this in the globals config. Was my solution of changing G28 to G28.6245197 acceptable?

**Reply by jschuh** [reply]
Yes, I assume those are getting called from a homing_override section, so calling G28.6245197 prevents klipper from failing on a recursive macro call.

**Reply by natetheskate** [reply]
Excellent! Thank you again.

**Comment by tastyratz** [comment]
I just went sensorless and klicky from a microswitch and omron sensor.

I'm running into the same error. I fixed the sensorless.cfg so no more errors, but, Now I'm getting errors with:

https://github.com/jlas1/Klicky-Probe/blob/main/Klipper_macros/klicky-macros.cfg

I renamed all the G28 commands to G28.6245197 in that file as well.

Does this workaround only work with 1 overlapping macro? I.E. I can only use G28.6245197 in once but that is now recursively failing out on the klicky-macros.cfg file?

Or is there something else in that file which is conflicting but showing up as G28 in the status?

```
2:14 PM
Malformed command 'SET_GCODE_VARIABLE MACRO=check_km_config VARIABLE=load_errors VALUE="['G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28', 'G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28', 'G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28']"'
2:14 PM
Malformed command 'SET_GCODE_VARIABLE MACRO=check_km_config VARIABLE=load_errors VALUE="['G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28', 'G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28', 'G28 in [homing_override] gcode. Replace with G28.6245197 to fix recursive macro call.\nSee readme: https://github.com/jschuh/klipper-macros#g28']"'
```

**Comment by tastyratz** [comment]
In case anyone else finds this later, I fixed the issue.

The klicky z tilt adjust cfg needed to be changed from G28 as well as the Klicky Macros file. The error message isn't just catching on G28 commands but also catches on the notifications in the klicky macros even though it's just text. You need to change *all* instances of G28 text, not just ones actively used as a command.
